sdk/core/Azure.Core/tests/BearerTokenAuthenticationPolicyTests.cs

```diff
@@ -439,22 +452,21 @@ public async Task BearerTokenAuthenticationPolicy_TokenExpiredThenFailed()
         }
 
         [Test]
-        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/14612")]
         public async Task BearerTokenAuthenticationPolicy_TokenAlmostExpiredThenFailed()
         {
             var requestMre = new ManualResetEventSlim(true);
             var responseMre = new ManualResetEventSlim(true);
             var credentialMre = new ManualResetEventSlim(false);
 
-            var getTokenRequestTimes = new List<DateTimeOffset>();
+            var getTokenRequestTimes = new ConcurrentQueue<DateTimeOffset>();
             var transportCallCount = 0;
             var credential = new TokenCredentialStub((r, c) =>
             {
                 if (transportCallCount > 0)
                 {
                     credentialMre.Set();
-                    getTokenRequestTimes.Add(DateTimeOffset.UtcNow);
-                    throw new InvalidOperationException("Error");
+                    getTokenRequestTimes.Enqueue(DateTimeOffset.UtcNow);
```

Comment:
Here list could be modified concurrently, so replacing it with thread-safe collection